python/lsst/ts/ess/controller/device/rpi_serial_hat.py

```diff
@@ -106,15 +107,19 @@ async def readline(self) -> str:
             the readline was started during device reception.
         """
         terminator = self.sensor.terminator.encode(self.sensor.charset)
-        bytes_read = await self.ser.read_until_async(expected=terminator)
-        st = self.sensor.terminator
+        bytes_read = await asyncio.wait_for(
+            self.ser.read_until_async(expected=terminator), timeout=5.0
+        )
 
         # Reading the first line of telemetry from the sensor may lead to
         # decoding errors since the line may have only been partially read. In
         # such case the decoding error is silently ignored.
-        if self.first_telemetry_read:
+        if not self.first_telemetry_read:
```

Comment:
Please update the above comment to match the code (which *always* ignores the first line read).

I also very gently suggest stripping the terminator in this method, but I realize it requires changes to the calling code and may not be worth the trouble. A few small advantages:
* It avoids the need to strip the terminator or deal with it at all. For example you can test if this is the first read by checking against "" instead of the terminator.
* It makes this code more like the ts_tcpip classes.